tests/components/lcn/conftest.py

```diff
@@ -104,9 +103,23 @@ def create_config_entry_myhome():
 
 async def init_integration(hass, entry):
     """Set up the LCN integration in Home Assistant."""
+    lcn_connection = MockPchkConnectionManager(
+        entry.data[CONF_IP_ADDRESS],
+        entry.data[CONF_PORT],
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        settings={
+            "SK_NUM_TRIES": entry.data[CONF_SK_NUM_TRIES],
+            "DIM_MODE": pypck.lcn_defs.OutputPortDimMode[entry.data[CONF_DIM_MODE]],
+        },
+        connection_id=entry.entry_id,
+    )
+
     entry.add_to_hass(hass)
-    await hass.config_entries.async_setup(entry.entry_id)
-    await hass.async_block_till_done()
+    with patch("pypck.connection.PchkConnectionManager", return_value=lcn_connection):
```

Comment:
It may be enough to just set `MockPchkConnectionManager` as side effect here.
```suggestion
    with patch("pypck.connection.PchkConnectionManager", side_effect=MockPchkConnectionManager) as manager:
```

Reply:
I guess that does not work? Setting the `side_effect` will not populate the `return_value` with the instance of `MockPchkConneciton` but with a `MagicMock`.

Reply:
The return value of the side effect will be set as return value:
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock

Reply:
That's strange... I don't know what I'm doing wrong here but if I implement your suggestion, `manager.return_value` indeed is a `MagicMock` (although it shouldn't).

```python
    with patch(
        "pypck.connection.PchkConnectionManager", side_effect=MockPchkConnectionManager
    ) as manager:
        await hass.config_entries.async_setup(entry.entry_id)
        await hass.async_block_till_done()

        print(manager.return_value)
        print(type(manager.return_value))
```
The print statements give:
```
<MagicMock name='PchkConnectionManager()' id='140468499110640'>
<class 'unittest.mock.MagicMock'>
```
I'm stumped.

Reply:
Maybe we need a lambda that creates the instance?

```py
side_effect=lambda *args: MockPchkConnectionManager(*args)
```

Reply:
That doesn't work either. Same result. Still confused.

Reply:
I dug into this a little bit.
Actually `return_value` is only set to the result of `side_effect` if the mock object returned from the context manager is called. Calling `side_effect` from within the context does not set `return_value` and therefore it sticks to the default `MagicMock`.

Well, what do you think of something like this (actually this is also a hack):
```python
temp = []
    
def create_mock(*args, **kwargs):
    temp.append(MockPchkConnectionManager(*args, **kwargs))
    return temp[0]

with patch("pypck.connection.PchkConnectionManager", side_effect=create_mock):
    ...

mock = temp[0]
```

Reply:
This is ok and better as we're not passing arguments in place of the code under test in the tests. We're still relying on the code under test to pass the arguments to the class which is what we want.

Reply:
Ok, I implemented this by using a nonlocal statement instead of the workaround with the list.